pkg/store/cache/caching_bucket.go

```diff
@@ -132,35 +221,221 @@ func (cb *CachingBucket) ReaderWithExpectedErrs(expectedFunc objstore.IsOpFailur
 	return cb.WithExpectedErrs(expectedFunc)
 }
 
-var chunksMatcher = regexp.MustCompile(`^.*/chunks/\d+$`)
+func (cb *CachingBucket) Iter(ctx context.Context, dir string, f func(string) error) error {
+	cfgName, cache, cfg := cb.findCacheConfig(cb.operationConfigs[operationIter], dir)
+	if cfg == nil {
+		return cb.Bucket.Iter(ctx, dir, f)
+	}
+	iterCfg := cfg.(iterConfig)
+
+	cb.operationRequests.WithLabelValues(operationIter, cfgName).Inc()
+
+	key := cachingKeyIter(dir)
+
+	data := cache.Fetch(ctx, []string{key})
+	if data[key] != nil {
+		list, err := decodeIterResult(data[key])
+		if err == nil {
+			cb.operationHits.WithLabelValues(operationIter, cfgName).Inc()
+
+			for _, n := range list {
+				err = f(n)
+				if err != nil {
+					return err
+				}
+			}
+			return nil
+		} else {
+			// This should not happen.
```

Comment:
mhm why? memcached can malform things, we could create incompatible encodings etc...

Reply:
I mean under normal conditions. Removed comment.